CMakeLists.txt

```diff
@@ -13,6 +22,7 @@ OPTION(USE_SWIG "Enable SWIG to generate Java API" OFF)
 OPTION(USE_HDFS "Enable HDFS support (EXPERIMENTAL)" OFF)
 OPTION(USE_R35 "Set to ON if your R version is not earlier than 3.5" OFF)
 OPTION(USE_TIMETAG "Set to ON to output time costs" OFF)
+OPTION(USE_CUDA "Enable CUDA-accelerated training" OFF)
```

Comment:
Maybe adding `(EXPERIMENTAL)` as the CUDA implementation is preliminary

Reply:
+1 for this. Besides all other factors, I see that CMake in recent versions is making a lot of progress in better CUDA support. Refer for instance to https://cmake.org/cmake/help/v3.18/release/3.18.html and https://cmake.org/cmake/help/v3.18/release/3.17.html. This note will allow us to bump the minimum required CMake version frequently without any concerns.